MergerLogic/ImageProcessing/TileMerger.cs

```diff
@@ -48,50 +41,38 @@ public TileMerger(ITileScaler tileScaler, IImageFormatter imageFormatter)
                         {
                             mergedImage.ColorSpace = ColorSpace.sRGB;
                             mergedImage.ColorType = mergedImage.HasAlpha ? ColorType.TrueColorAlpha : ColorType.TrueColor;
-                            this._imageFormatter.CovertToFormat(mergedImage,format);
+                            this._imageFormatter.CovertToFormat(mergedImage, format);
                             var mergedImageBytes = mergedImage.ToByteArray();
+
+                            for (var i = images.Count - 1; i >= 0; i--)
+                            {
+                                images[i].Dispose();
+                            }
+
                             return mergedImageBytes;
                         }
+
                     }
             }
         }
 
-        private List<MagickImage> GetImageList(List<CorrespondingTileBuilder> tiles, Coord targetCoords,
-            out Tile? lastProcessedTile, out bool singleImage)
+        private List<MagickImage> GetImageList(List<CorrespondingTileBuilder> tiles, Coord targetCoords, out bool singleImage)
         {
             var images = new List<MagickImage>();
-            lastProcessedTile = null;
             int i = tiles.Count - 1;
             Tile? tile = null;
 
             singleImage = false;
             bool hasAlpha = false;
             try
             {
-                tile = GetFirstTile(tiles, targetCoords, ref lastProcessedTile, ref i);
-                for (; i >= 0; i--)
+                tile = GetFirstTile(tiles, targetCoords, ref i);
+                
+                this.AddTileToImageList(targetCoords, tile, images, out hasAlpha);
+                if (!hasAlpha)
```

Comment:
please use _ImageUtils.IsTransparent(scaledImage)_ for PNG's (only!)

Reply:
Changed.